packages/namers/default/src/DefaultNamer.js

```diff
@@ -13,24 +12,7 @@ const COMMON_NAMES = new Set(['index', 'src', 'lib']);
 export default (new Namer({
   name({bundle, bundleGraph, options}) {
     let bundleGroup = bundleGraph.getBundleGroupsContainingBundle(bundle)[0];
-    let bundleGroupBundles = bundleGraph.getBundlesInBundleGroup(bundleGroup);
-
-    if (bundle.isEntry) {
-      let entryBundlesOfType = bundleGroupBundles.filter(
-        b => b.isEntry && b.type === bundle.type,
-      );
-      assert(
-        entryBundlesOfType.length === 1,
-        // Otherwise, we'd end up naming two bundles the same thing.
-        'Bundle group cannot have more than one entry bundle of the same type',
```

Comment:
Do we want to keep this error in some form? cc @wbinnssmith

Reply:
I rebased this branch and  preserved this assertion, since we're not sure if we still need it or not.